### PR TITLE
Show a Toast for HMR

### PR DESCRIPTION
**Pull request by satya164**
This is just to try out the experience with HMR if we show toasts (might be annoying). Let's try it out before deciding on merging/closing.

Related #5906

### CODE REVIEW

**Comment by janicduplessis** [comment]
You got an extra commit 'Annotate WebSocket with Flow' in this PR (I always do that too :))

**Comment by satya164** [comment]
@janicduplessis Yeah, I sent that as a PR to master, but isn't merged yet :(

**Comment by janicduplessis** [comment]
You can rebase -i and delete the commit.

**Comment by satya164** [comment]
@janicduplessis I need that commit. Otherwise Flow will give me error.

**Comment by janicduplessis** [comment]
Ahhh ok that's good then.

**Comment by skevy** [comment]
Code looks good to me @satya164...I'll try to try this out later today to see how it feels.

**Comment by satya164** [comment]
@skevy Awesome :)

File: Libraries/Utilities/HMRLoadingView.android.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule HMRLoadingView
+ * @flow
+ */
+
+'use strict';
+
+const ToastAndroid = require('ToastAndroid');
+const TOAST_SHORT_DELAY = 2000;
```
Comment (martinbigio) [inline Libraries/Utilities/HMRLoadingView.android.js]: FYI on iOS we minimum amount the header notification is shown is 600ms. We could have a different one on Android if it feels better though

Comment (satya164) [inline Libraries/Utilities/HMRLoadingView.android.js]: @martinbigio We can't change it, unless we have the ability to cancel the toast (which is still flaky, but still)

Comment (satya164) [inline Libraries/Utilities/HMRLoadingView.android.js]: We can only choose between 2s and 3.5s for Toasts.

Comment (martinbigio) [inline Libraries/Utilities/HMRLoadingView.android.js]: Would be great being able to cancel the toast. HMR updates usually take 500ms :).

Comment (satya164) [inline Libraries/Utilities/HMRLoadingView.android.js]: @martinbigio Yeah, PR is here - #5906

**Comment by satya164** [comment]
@skevy Did you try this?

**Comment by martinbigio** [comment]
I think we should be able to enable HMR on the 0.22 cut but among other things we some kind of visual indicator in, like what @satya164 implemented here. Would be awesome being able to dismiss the indicator (5906)

**Comment by satya164** [comment]
@martinbigio I'll do some testing with both PRs and will try to get this merged ASAP.

**Comment by martinbigio** [comment]
Thanks @satya164!

**Comment by satya164** [comment]
Merging this for now. Currently the Toast is shown for short period (2s) and doesn't seem to annoy much. Let's come up with better plans if we wanna change.

**Comment by satya164** [comment]
@facebook-github-bot shipit

**Comment by satya164** [comment]
@facebook-github-bot shipit

**Comment by satya164** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**